pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -421,7 +421,11 @@ impl ConnectionManagerState {
                     Err(e) => {
                         use super::walreceiver_connection::ExpectedError;
                         if e.is_expected() {
-                            info!("walreceiver connection handling ended: {e:#}");
+                            if let Some(adjusted_error) = e.maybe_reformat() {
+                                info!("walreceiver connection handling ended: {adjusted_error}");
```

Comment:
Duplicating this format string is a little ugly, but avoids externally String-izing the `e` to make it usable in a `unwrap_or` on the maybe_reformat.  Not sure if there's a trick way to do this in one line such that the `info!` macro can take either the `Error` or a `String`.

Reply:
Aah, when looking this in a call I missed that this was a method in the extension trait.

It might be good idea future-wise to add now checking for is this the particular message safekeepers send when there's no running compute and we've caught up, then we can just decide to log our own static string.

While doing this, it would be worthwhile to look up the origin for the message in safekeeper side, and add a comment to both sides so that we will have a very obvious warning for changing anything regarding this.

Reply:
Unsure if the way I did it on line R459-R460 is a good enough way, but doubtful if we should have only the successful completion either.

Aah but these are overlapping now, failed to realize it. Yeah, I dislike how this now looks since the checks are overlapping, but not completly.

Reply:
This is all in the WalReceiverError type now, and added a commit to put comments in the safekeeper to protect the magic string.